commit 8c6cd83a4825155d1ee9ddcd29b023682944e3e6
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Wed Mar 12 15:41:30 2014 +0100

    Fix traceback when history files don't exist and user is not root.

diff --git a/yum/history.py b/yum/history.py
index 6f60f54..3f20128 100644
--- a/yum/history.py
+++ b/yum/history.py
@@ -668,6 +668,7 @@ class YumHistory:
 
         self.releasever = releasever
 
+        self._db_file = None
         if not os.path.exists(self.conf.db_path):
             try:
                 os.makedirs(self.conf.db_path)
@@ -680,7 +681,6 @@ class YumHistory:
                 self.conf.writable = True
 
         DBs = glob.glob('%s/history-*-*-*.sqlite' % self.conf.db_path)
-        self._db_file = None
         for d in reversed(sorted(DBs)):
             fname = os.path.basename(d)
             fname = fname[len("history-"):-len(".sqlite")]
diff --git a/yumcommands.py b/yumcommands.py
index 4e72a71..75b3ce2 100644
--- a/yumcommands.py
+++ b/yumcommands.py
@@ -3051,7 +3051,7 @@ class HistoryCommand(YumCommand):
         if extcmds and extcmds[0] in ('repeat', 'redo', 'undo', 'rollback', 'new'):
             checkRootUID(base)
             checkGPGKey(base)
-        elif not os.access(base.history._db_file, os.R_OK):
+        elif not (base.history._db_file and os.access(base.history._db_file, os.R_OK)):
             base.logger.critical(_("You don't have access to the history DB."))
             raise cli.CliError
 
